src/main/java/org/generationcp/commons/workbook/generator/GermplasmAttributesWorkbookExporter.java

```diff
@@ -1,38 +1,52 @@
 package org.generationcp.commons.workbook.generator;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.poi.ss.usermodel.CellStyle;
 import org.generationcp.commons.parsing.ExcelCellStyleBuilder;
-import org.generationcp.middleware.manager.api.GermplasmDataManager;
-import org.generationcp.middleware.pojos.UserDefinedField;
+import org.generationcp.middleware.ContextHolder;
+import org.generationcp.middleware.domain.ontology.Variable;
+import org.generationcp.middleware.domain.ontology.VariableType;
+import org.generationcp.middleware.manager.ontology.api.OntologyVariableDataManager;
+import org.generationcp.middleware.manager.ontology.daoElements.VariableFilter;
+import org.generationcp.middleware.util.VariableValueUtil;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import javax.annotation.Resource;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Component
-public class GermplasmAttributesWorkbookExporter extends GermplasmAddedColumnExporter<UserDefinedField> {
+public class GermplasmAttributesWorkbookExporter extends GermplasmAddedColumnExporter<Variable> {
 
-	@Resource
-	private GermplasmDataManager germplasmManager;
+	@Autowired
+	private OntologyVariableDataManager ontologyVariableDataManager;
 
 	private List<String> addedAttributeColumns = new ArrayList<>();
 
 	@Override
-	List<UserDefinedField> getSourceItems() {
-		final List<UserDefinedField> attributeTypeColumns = new ArrayList<>();
+	List<Variable> getSourceItems() {
+		final List<Variable> attributeTypeColumns = new ArrayList<>();
 		//columnsInfo is null when exporting germplasm list from Study Manager
 		if (this.columnsInfo != null) {
-			final List<UserDefinedField> attributeTypes = this.germplasmManager.getAllAttributesTypes();
-			final Map<String, UserDefinedField>
-				attributeTypesMap = attributeTypes.stream().collect(Collectors.toMap(u -> u.getFcode().toUpperCase(), u -> u, (u1, u2) -> u1));
+			final VariableFilter variableFilter = new VariableFilter();
```

Comment:
There are several places here in Workbench and also in Middleware where we have same code to retrieve passport/attribute variables. Better to centralize in `GermplasmService` or `GermplasmAttributeService` a method:
`List<Variable getAttributeAndPassportVariables(String programUUID)`.

Reply:
I think that this is not really needed, this service will end calling ontologyDataManager to get the information
What I think we should have is a new VariableService in middleware but I will leave out of the scope of this ticket as of now